stan/math/prim/mat/err/check_finite.hpp

```diff
@@ -29,14 +31,15 @@ struct finite<Eigen::Matrix<T, R, C>, true> {
                     const Eigen::Matrix<T, R, C>& y) {
     if (!value_of(y).allFinite()) {
       for (int n = 0; n < y.size(); ++n) {
-        if (!(boost::math::isfinite)(y(n))) {
+        if (!std::isfinite(value_of_rec(y(n)))) {
```

Comment:
So this could only potentially have an impact on performance if we have non-finite values in y. In which case performance wont be an issue anyways as this will throw. I do not see why this would affect performance anyway.